build.sbt

```diff
@@ -9,16 +9,24 @@ lazy val commonSettings = Seq(
 
 lazy val root = (project in file(".")).
   settings(commonSettings).
-  aggregate(library)
+  aggregate(library, plugin)
 
-lazy val library = project.
+lazy val plugin = (project in file("plugin")).
   settings(commonSettings).
   settings(
-    name := "datatype",
-    libraryDependencies ++= jsonDependencies ++ Seq(specs2 % Test),
-    publishArtifact in (Test, packageBin) := true, // for scripted + SchemaExample
+    sbtPlugin := true,
+    name := "datatype-plugin",
+    ScriptedPlugin.scriptedSettings,
     scriptedLaunchOpts := { scriptedLaunchOpts.value ++
       Seq("-Xmx1024M", "-XX:MaxPermSize=256M", "-Dplugin.version=" + version.value)
-    }
+    },
+    publishLocal <<= (publishLocal) dependsOn (publishLocal in library)
```

Comment:
@eed3si9n This is required to have `library` being locally published before the scripted tests are run. Shouldn't it be done automatically? If not, is this the right way to achieve this behaviour?